packages/internal/src/generate/plugins/rw-typescript-resolvers/visitor.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  TypeScriptResolversPluginConfig,
+  TypeScriptResolversVisitor,
+} from '@graphql-codegen/typescript-resolvers'
+import {
+  indent,
+  DeclarationBlock,
+} from '@graphql-codegen/visitor-plugin-common'
+import {
+  FieldDefinitionNode,
+  GraphQLSchema,
+  ObjectTypeDefinitionNode,
+} from 'graphql'
+
+export class RwTypeScriptResolversVisitor extends TypeScriptResolversVisitor {
+  constructor(
+    pluginConfig: TypeScriptResolversPluginConfig,
+    schema: GraphQLSchema
+  ) {
+    super(pluginConfig, schema)
+  }
+
+  FieldDefinition(
+    node: FieldDefinitionNode,
+    key: string | number,
+    parent: any
+  ): (parentName: string) => string | null {
+    const hasArguments = node.arguments && node.arguments.length > 0
+
+    const superFieldDefinition = super.FieldDefinition(node, key, parent)
+
+    return (parentName: string) => {
+      // We're reusing pretty much all of the logic in the original plugin
+      // Visitor implementation by calling the `super` method here
+      const fieldDef = superFieldDefinition(parentName)
+
+      // If this field doesn't take any arguments, and it is a resolver, then
+      // we switch to using the OptArgsResolver type instead, so that the user
+      // isn't forced to pass in arguments that aren't going to be used anyway
+      if (!hasArguments && fieldDef?.includes(': Resolver<')) {
```

Comment:
Remember the name "Resolver" can be changed with a config - I think maybe you can load the value from config - which by default should be "Resolver"

Reply:
I don't think you can change the "Resolver" name. It looks to be hard coded here: https://github.com/dotansimha/graphql-code-generator/blob/c6c60a3078f3797af435c3852220d8898964031d/packages/plugins/other/visitor-plugin-common/src/base-resolvers-visitor.ts#L1030

(`this._directiveResolverMappings[...]` a few lines above returns `[]` in our case)